### PR TITLE
BLD: rasterizer in PyTorch

### PR DESCRIPTION
**Pull request by eigenfoo**

### CODE REVIEW

**Comment by eigenfoo** [comment]
@ArianaFreitag @ccurro I think this PR is ready for review.

- `base` and `base_half` have speed scaling implemented.
- `shrunk`, `bounded` and `tiled` have all been commented out (for now?). If we prefer, I can just delete the commented code to make it easier to read. If necessary, we can implement speed scaling for the other raster methods too, but that should be a subsequent PR.
- I _think_ we can make some performance improvements, but again, that deserves to be its own PR.

**Comment by ArianaFreitag** [comment]
Overall review of this PR is that the added methods for calculating the derivatives make sense and seem really clear. My main confusion I have has to deal with some the variable names in the doc strings, particularly num_curves and num_bezier, and their relation. Other than that the code looked fine to me and i easily see where the new methods fit into the old code. i say merge after we make the doc string variables uniform

File: src/rasterizer/rasterizer.py [hunk 1]
```diff
@@ -0,0 +1,580 @@
...
+
+    Returns
+    -------
+    samples : [2, num_steps]
+    '''
+    P0 = control_points[0].unsqueeze(1)
+    P1 = control_points[1].unsqueeze(1)
+    P2 = control_points[2].unsqueeze(1)
+    samples = (P1 + (1 - steps)**2 * (P0 - P1) + steps**2 * (P2 - P1))
+    return samples
```
Comment (ccurro) [inline src/rasterizer/rasterizer.py]: I have a preference for an empty line before the return line. I thought yapf did as well, but it seems it doesn't.

Comment (eigenfoo) [reply src/rasterizer/rasterizer.py]: I don't think any formatter has a preference. [Black](https://github.com/ambv/black) also doesn't have a preference, for instance. In any case it's a simple enough function I think.

**Comment by eigenfoo** [comment]
I wasn't really familiar with the `.to(device)` stuff. I'll give it a try soon.

**Pull request merged**